src/Events/Editor/Full_Site/Templates.php

```diff
@@ -73,26 +77,84 @@ public function add_events_archive( array $query_result, $query, string $templat
 	 * Returns the constructed template object for the query.
 	 *
 	 * @since 5.14.2
+	 * @since TBD Updated with WP_Block_Template initialization from the database for Site Editor.
 	 *
 	 * @return WP_Block_Template A reference to the template object for the query.
 	 */
 	public function get_template_events_archive() {
-		$template_content = file_get_contents(
-			Tribe__Events__Main::instance()->plugin_path . '/src/Events/Editor/Full_Site/Templates/archive-events.html'
-		);
+		$template      = new WP_Block_Template();
+		$archive_block = tribe( Archive_Events::class );
+
+		// Let's see if we have a saved template?
+		$wp_query_args  = [
+			'post_name__in'  => [ $archive_block->slug() ],
+			'post_type'      => 'wp_template',
+			'post_status'    => [ 'auto-draft', 'draft', 'publish', 'trash' ],
+			'posts_per_page' => 1,
+			'no_found_rows'  => true,
+			'tax_query'      => [
+				[
+					'taxonomy' => 'wp_theme',
+					'field'    => 'name',
+					'terms'    => $archive_block->get_namespace(),
+				],
+			],
+		];
+		$template_query = new WP_Query( $wp_query_args );
+		$posts          = $template_query->posts;
+
+		// If empty, this is our first time loading our Block Template. Let's create it.
+		if ( empty( $posts ) ) {
+			$insert = [
+				'post_name'    => $archive_block->slug(),
+				'post_title'   => esc_html_x( 'Calendar Views (Event Archive)', 'The Full Site editor block navigation title', 'the-events-calendar' ),
+				'post_excerpt' => esc_html_x( 'Displays the calendar views.', 'The Full Site editor block navigation description', 'the-events-calendar' ),
+				'post_type'    => 'wp_template',
+				'post_status'  => 'publish',
+				'post_content' => Template_Utils::inject_theme_attribute_in_content( file_get_contents(
+					Tribe__Events__Main::instance()->plugin_path . '/src/Events/Editor/Full_Site/Templates/archive-events.html'
+				) )
+			];
+			// Create this template.
+			$id = wp_insert_post( $insert );
+
+			// Setup our "theme" term, for the taxonomy query.
+			$term = get_term_by( 'name', $archive_block->get_namespace(), 'wp_theme', ARRAY_A );
+			if ( ! $term ) {
+				wp_insert_term( $archive_block->get_namespace(), 'wp_theme' );
+			}
+			wp_set_post_terms( $id, $archive_block->get_namespace(), 'wp_theme' );
```

Comment:
```suggestion
			wp_set_post_terms( $id, $archive_block->get_namespace(), 'wp_theme' );
```
@EugeneKyale just thought of this while working on single. Does it make sense for us to store `tec` as a `wp_theme`? I am not confident in the approach here. Is there a precedence for this you know of?

Reply:
I took this approach because WP core attempts to hydrate the template via `_build_block_template_result_from_post()` - and this is how it structures the template post query.

Reply:
@stratease I'm not aware of any precedence and don't have a strong opinion either way. 

A general thought I have is that if `wp_theme` is being used by other themes or plugins, it might be better to use a more specific custom taxonomy to avoid potential conflicts.

Reply:
@EugeneKyale it's not a personal choice, it's just based on how the [WP core will look for the block template](https://github.com/WordPress/WordPress/blob/master/wp-includes/block-template-utils.php#L1055-L1068) in a couple places like `get_block_template()` - it is hard coded to use that taxonomy. My concern is maybe I'm using the wrong thing, couldn't find any detailed articles/docs on setting this up, they were pretty high level and didn't touch some of this.

@bordoni do you have thoughts here?

Reply:
Also worth noting [how it hydrates the template](https://github.com/WordPress/WordPress/blob/master/wp-includes/block-template-utils.php#L739-L761)